index.js

```diff
@@ -190,56 +190,97 @@ class ServerlessCustomDomain {
+      throw Error(`Could not find the certificate ${certificateName}`);
+    }
+    return certificateArn;
+  }
+
+  /*
    * Obtains the certification arn
    */
   getCertArn() {
     const acm = new AWS.ACM({
       region: 'us-east-1',
     });       // us-east-1 is the only region that can be accepted (3/21)
 
-    const certArn = acm.listCertificates().promise();
+    const issuedCertArn = acm.listCertificates({ CertificateStatuses: ['ISSUED'] }).promise();
 
-    return certArn.then((data) => {
-      // The more specific name will be the longest
-      let nameLength = 0;
-      // The arn of the choosen certificate
-      let certificateArn;
-      // The certificate name
-      let certificateName = this.serverless.service.custom.customDomain.certificateName;
-
-
-      // Checks if a certificate name is given
-      if (certificateName != null) {
-        const foundCertificate = data.CertificateSummaryList
-          .find(certificate => (certificate.DomainName === certificateName));
+    return issuedCertArn.then((data) => {
+      if (process.env.SLS_DEBUG && Array.isArray(data.CertificateSummaryList)) {
+        this.serverless.cli.log(`Found ${data.CertificateSummaryList.length} Valid Certificates: ${JSON.stringify(data.CertificateSummaryList)}`);
+      }
+      return this.findClosetCertificate(data);
+    }).catch((err) => {
+      // No cert found in the ISSUED status. Let's look through certs in all other statuses.
+      const certArn = acm.listCertificates().promise();
 
-        if (foundCertificate != null) {
-          certificateArn = foundCertificate.CertificateArn;
+      return certArn.then((data) => {
+        if (process.env.SLS_DEBUG && Array.isArray(data.CertificateSummaryList)) {
+          this.serverless.cli.log(`Found ${data.CertificateSummaryList.length} Certificates: ${JSON.stringify(data.CertificateSummaryList)}`);
         }
-      } else {
-        certificateName = this.givenDomainName;
-        data.CertificateSummaryList.forEach((certificate) => {
-          let certificateListName = certificate.DomainName;
-
-          // Looks for wild card and takes it out when checking
-          if (certificateListName[0] === '*') {
-            certificateListName = certificateListName.substr(1);
-          }
-
-          // Looks to see if the name in the list is within the given domain
-          // Also checks if the name is more specific than previous ones
-          if (certificateName.includes(certificateListName)
-            && certificateListName.length > nameLength) {
-            nameLength = certificateListName.length;
-            certificateArn = certificate.CertificateArn;
-          }
-        });
-      }
 
-      if (certificateArn == null) {
-        throw Error(`Could not find the certificate ${certificateName}`);
-      }
-      return certificateArn;
+        this.findClosetCertificate();
+      }, () => {
+        // rethrow the original error
+        throw err;
```

Comment:
Won't this happen automatically if we just don't provide a failure function?

Reply:
Great observation, after looking closely I noticed a missing return and the re-throw on the second .then, line 275, was exactly as describe.

However the noted line above should remain. The reason is the "catch" that returns a promise could fail for a multitude of reasons. On line 254 the returned promise  is attempting to get a list of all certs. For example the error could be something from the AWS framework like "Unable to get a list of all certificates" or "Unauthorized".

The error that the user cares about is the previous error from not finding the cert on line 245.  `Could not find the certificate ${certificateName}`.

Since the catch method is simply there to get a more specific error I would suggest re-throwing the err from line 250 since it is the error the user would care most about. I don't think they care if something went wrong while trying to get locate the cert in an invalid status.